Dev hyun

hyunsir opened the pull request:
1.重新写了一下运行代码，batch_run.py  batch_rerun.py可以在终端运行
2.写了几个关于 *.bin文件 有关的测试
3.collection现在可以做加法
(由于1和3 ，现在分批跑可以合并，并且可以检查有多少没跑成功，逐一tag重新跑，不会浪费时间)

hyunsir committed:
重新写了一下运行代码
写了几个关于bin内容有关的测试
collection现在可以做加法
(分批跑可以合并，并且可以检查有多少没跑成功，逐一tag重新跑，不会浪费时间)

hyunsir committed:
更新过滤规则

mingwei-liu commented on batch_rerun.py:
```diff
@@ -0,0 +1,32 @@
+from sotag.models.tag import SOTagItem
+
+
+def rerun_batch(sta: int, end: int):
```
最好在根目录底下，建一个script 目录，然后这些可以执行脚本放到底下，参考我们其他项目的设计，根目录底下最好不要出现代码

mingwei-liu replied:
tags.xml 这个文件不应该加到git管理吧？这个文件应该在zip压缩包里面

mingwei-liu commented on path.py:
```diff
@@ -0,0 +1,3 @@
+import os
```
这个文件的名字最好不要叫path，参考KG项目模板的名字。因为path是个常用词，容易冲突

mingwei-liu commented on sotag/download/sotag_downloader.py:
```diff
@@ -248,7 +257,7 @@ def run(self):
         return self.so_tag_item_collection
 
-    def save(self, path):
+    def save(self, path: object) -> object:
```
这个函数没有返回值，path可以指定类型一定要是str或者Path对象

mingwei-liu commented on sotag/models/tag.py:
```diff
@@ -140,18 +145,28 @@ def get_so_tag_item(self, tag_name):
 
     def sub_collection(self, tag_names):
-        subcollection = SOTagItemCollection()
+        subcollection = SOTagItemCollection()  # 给定一个名字列 从原来collection中拷贝出一个子collection
```
中文这一句话可以写成方法注释

mingwei-liu commented on sotag/models/tag.py:
```diff
@@ -140,18 +145,28 @@ def get_so_tag_item(self, tag_name):
-        return set(self.name2sotag.keys()) | set(self.name2sotag.keys())
+        return set(self.name2sotag.keys()) | set(self.synonyms2tagname.keys())
+
+# class SOTagSynonymsCollection(SaveLoad):
```
注释掉的这些可以考虑删除

mingwei-liu commented on sotag/models/tag.py:
```diff
@@ -131,6 +131,11 @@ def add_so_tag_item(self, so_tag_item: SOTagItem):
+            self.add_so_tag_item(so_tag_item)
+        return self
+
     def get_so_tag_item(self, tag_name):
```
可以考虑给方法补齐注释和参数类型注解，返回值注解之类。

mingwei-liu commented on tests/download/test_sotag_downloader.py:
```diff
@@ -1,19 +1,79 @@
+
+    def test_show_one_tag_in_collection(self):
+        # 此函数用于测试 在本地的某个bin中 是否有 某个tag的全部SOTagItem类型中定义的数据 ，参数是一个地址 和一个tag字符串
+        searcher = SOTagDownloader(so_tag_item_collection_path="/{}/data/run_100_10000.bin".format(ROOT_PATH))
```
path最好用Path对象+/ 这种方式构造，不要用拼接的方式

mingwei-liu reviewed (changes requested):
需要改一改

hyunsir committed:
更新过滤规则

hyunsir committed:
更新过滤规则

hyunsir committed:
1.重新写了一下运行代码，batch_run.py batch_rerun.py可以在终端运行
2.写了几个关于 *.bin文件 有关的测试
3.collection现在可以做加法
(由于1和3 ，现在分批跑可以合并，并且可以检查有多少没跑成功，逐一tag重新跑，不会浪费时间)

mingwei-liu commented on tests/download/test_sotag_downloader.py:
```diff
@@ -1,19 +1,76 @@
+    def test_bin_tag_items_to_csv(self):
+        # 此函数用于将 本地某个bin中的数据 格式化保存到 本地csv，方便测试查看，一般不用
+        path = Path(ROOT_DIR)
+        bin_path = str(path / "data/so_tag_info2.bin")
```
这个不够彻底，有些时候还是会出问题,最好， str(path / "data"/"so_tag_info2.bin")

mingwei-liu commented on tests/download/test_sotag_downloader.py:
```diff
@@ -1,19 +1,76 @@
+        with open(csv_path, 'a+', newline='\n') as file:
+            csv_file = csv.writer(file)
+            for tag in collection.name2sotag.keys():
+                item: SOTagItem = collection.get_so_tag_item(tag)
```
可以给SOTag加个 to_dict 方法，将类对象数据转换成字典返回

hyunsir committed:
添加了一些测试使用的小型数据
